lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -763,6 +788,36 @@ void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAddTensorOp
+//===----------------------------------------------------------------------===//
+
+void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                  MLIRContext *context) {
+  patterns.add(+[](AtenAddTensorOp op, PatternRewriter &rewriter) {
+    // The lhs and rhs of the add.tensor op should be 0d tensors for the
+    // canonicalization to be carried out.
+    // `aten.add.tensor(self, other, alpha)` is canonicalized to
+    // `aten.add.int(self, aten.mul.int(other, alpha))`.
+
+    Value lhs = getScalarValue(op.self(), op.getLoc(), rewriter);
```

Comment:
you should check `if (!lhs)` here before the next call -- this avoids creating IR when a match fails.

Reply:
@silvasean I update that part of the code and just want to make sure this is the you wanted.

Reply:
Perfect :)